### ISSUE TITLE
Can't get FCM_KEY after upgrading to firebase_messaging: ^3.0.0

### ISSUE THREAD

**Issue by trey-rosius** [flutter/flutter#28114]
My application was working fine until i upgraded the firebase messaging to version 3.0.0. 

Here is the warning message i get
```
/Users/rosius/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-3.0.1/android/src/main/java/io/flutter/plugins/firebasemessaging/FlutterFirebaseInstanceIDService.java:11: warning: [deprecation] FirebaseInstanceIdService in com.google.firebase.iid has been deprecated
import com.google.firebase.iid.FirebaseInstanceIdService;
                              ^
/Users/rosius/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-3.0.1/android/src/main/java/io/flutter/plugins/firebasemessaging/FlutterFirebaseInstanceIDService.java:13: warning: [deprecation] FirebaseInstanceIdService in com.google.firebase.iid has been deprecated
public class FlutterFirebaseInstanceIDService extends FirebaseInstanceIdService {
                                                      ^
/Users/rosius/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-3.0.1/android/src/main/java/io/flutter/plugins/firebasemessaging/FlutterFirebaseInstanceIDService.java:20: warning: [deprecation] getToken() in FirebaseInstanceId has been deprecated
    intent.putExtra(EXTRA_TOKEN, FirebaseInstanceId.getInstance().getToken());
                                                                 ^
/Users/rosius/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-3.0.1/android/src/main/java/io/flutter/plugins/firebasemessaging/FlutterFirebaseInstanceIDService.java:26: warning: [deprecation] onTokenRefresh() in FirebaseInstanceIdService has been deprecated
  public void onTokenRefresh() {
              ^
4 warnings
```

**Here is the error message i get**

```
E/FirebaseInstanceId(26269): Token retrieval failed: SERVICE_NOT_AVAILABLE
W/FirebaseMessagingPlugin(26269): getToken, error fetching instanceID: 
W/FirebaseMessagingPlugin(26269): java.io.IOException: SERVICE_NOT_AVAILABLE
W/FirebaseMessagingPlugin(26269): 	at com.google.firebase.iid.zzr.zza(Unknown Source:66)
W/FirebaseMessagingPlugin(26269): 	at com.google.firebase.iid.zzr.zza(Unknown Source:79)
W/FirebaseMessagingPlugin(26269): 	at com.google.firebase.iid.zzu.then(Unknown Source:4)
W/FirebaseMessagingPlugin(26269): 	at com.google.android.gms.tasks.zzd.run(Unknown Source:5)
W/FirebaseMessagingPlugin(26269): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/FirebaseMessagingPlugin(26269): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/FirebaseMessagingPlugin(26269): 	at java.lang.Thread.run(Thread.java:764)
D/        (26269): HostConnection::get() New Host Connection established 0xe0b8abc0, tid 26319
D/EGL_emulation(26269): eglMakeCurrent: 0xe1173680: ver 3 0 (tinfo 0xdfed92d0)
I/flutter (26269): Push Messaging token: null
```

**Here's my flutter doctor**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.2.2-pre.21, on Mac OS X 10.13.6 17G65, locale en-CM)
[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    ✗ Android license status unknown.
[!] iOS toolchain - develop for iOS devices (Xcode 10.0)
    ✗ Verify that all connected devices have been paired with this computer in Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
[!] Android Studio (not installed)
[✓] VS Code (version 1.31.0)
[✓] Connected device (1 available)

! Doctor found issues in 3 categories.
```

**Comment by zoechi** [comment]
I assume you migrated to AndroidX, right?

Please check https://stackoverflow.com/questions/49949818/firebaseinstanceid-token-retrieval-failed-service-not-available/50517006

**Comment by trey-rosius** [comment]
I resolved this issue by doing two things. Firstly, i updated the google Play Services of the android emulator. Then i encountered this error
`
Connection closed before full header was received, flutter uri = http://127.0.0.1:58839/ws
`
and this was the culprit.**flutter_facebook_login: ^1.2.0** . This library hasn't been migrated to AndroidX yet. 
Then i went to the github profile of the library and found this 
https://github.com/roughike/flutter_facebook_login/pull/87
.I applied the suggested fix. 
**https://github.com/roughike/flutter_facebook_login/pull/87#issuecomment-463678640**
Then the app doesn't throw anymore exceptions, but closes immediately at start up.
Judging from all these, i can conclude that there is a compatibility issue between flutter_facebook_login 1.2.0 and 
firebase_messaging 3.0.1. 
So i've taken off the facebook library and now  i can get my FCM token, at the expense of the  facebook login(sad).
